src/components/Form/Form.tsx

```diff
@@ -0,0 +1,113 @@
+import { Button } from "@components/Button/Button";
+import { IconNames } from "@data/interfaces/Icons";
+import styles from "./Form.module.css";
+import Input from "@components/Input/Input";
+import { contactForm } from "@data/static/forms";
+import useApi from "@utils/useApi";
+import ReCAPTCHA from "react-google-recaptcha";
+import { useState } from "react";
+
+// Definir un tipo para las validaciones de formulario
+type FormValidationSchema = {
+  [key: string]: Yup.StringSchema;
+};
+interface FormProps {
+  onToastShow: (message: string, error: boolean) => void;
+}
+export const Form = ({ onToastShow }: FormProps) => {
+  const { loading, sendEmail } = useApi(); // Usar el hook useApi
+  const [isVerified, setIsVerified] = useState(false);
+
+  const handleRecaptchaChange = (value: string | null) => {
+    console.log("Recaptcha value:", value);
+    setIsVerified(!!value);
+  };
+
+  // Construir el esquema de validación del formulario
+  const validationSchema: FormValidationSchema = contactForm.fields.reduce(
+    (acc, field) => {
+      acc[field.name] = field.validation;
+      return acc;
+    },
+    {} as FormValidationSchema,
+  );
+
+  const formik = useFormik({
+    initialValues: Object.fromEntries(
+      contactForm.fields.map((field) => [field.name, ""]),
+    ),
+    validationSchema: Yup.object().shape(validationSchema),
+    onSubmit: async (values) => {
+      try {
+        if (!isVerified) {
+          console.error("Por favor, verifica el reCAPTCHA.");
+          onToastShow("Por favor, verifica el reCAPTCHA.", true);
+
+          return;
+        }
+
+        await sendEmail({
+          userEmail: values.emailInput,
+          userName: values.nameInput,
+          userMessage: values.messageInput,
+          userBusiness: values.businessInput,
+          userPhone: values.cellphoneNumber,
+        });
+        onToastShow("Correo electrónico enviado exitosamente.", false);
+      } catch (error) {
+        console.error("Error al enviar el correo electrónico:", error);
+        onToastShow("Error al enviar el correo electrónico.", true);
+      }
+    },
+  });
+
+  return (
+    <form onSubmit={formik.handleSubmit} className={styles.formWrapper}>
+      <div className={styles.inputs}>
+        {contactForm.fields.map((field, index) => (
+          <Input
+            key={index}
+            type={field.type}
+            name={field.name}
+            placeholder={field.placeholder}
+            label={field.label}
+            required={field.required}
+            hidden={field.hidden}
+            onChange={formik.handleChange}
+            onBlur={formik.handleBlur}
+            value={formik.values[field.name]}
+            error={
+              formik.touched[field.name] &&
+              (formik.errors[field.name] as string)
+            }
+          />
+        ))}
+      </div>
+      <div className={styles.recaptcha}>
+        <ReCAPTCHA
+          sitekey={"6Lc4NYIpAAAAAH_XFbEFTtrLMMVPEbu6cffo2gs0"}
```

Comment:
⚠️⚠️⚠️
Security leak here. We cannot have this kind of keys in the repo -- which is public btw. 

You must re-generate the key and we need to create a .env and .env.example files in the app root dir.
Hide the .env file in the .gitignore so the file is not commitable

**.env** 
// this file is not going to be pushed into the repo
```bash
VITE_RECAPTCHA_KEY=32132
```

**.env.example** 
// this file is going to be pushed to the repo as example of the required env variables
// all key values are removed or completed with dummy data 
```bash
VITE_RECAPTCHA_KEY=your_key
```